### ISSUE TITLE
Explain database session storage config in Flex

### ISSUE THREAD

**Issue by aik099** [symfony/symfony-docs#8420]
According to docs on https://symfony.com/doc/current/doctrine/pdo_session_storage.html page to avoid duplication the `PdoSessionHandler` needs to be configured like this:

```yml
services:
    # ...

    Symfony\Component\HttpFoundation\Session\Storage\Handler\PdoSessionHandler:
        public:    false
        arguments:
            - 'mysql:host=%database_host%;port=%database_port%;dbname=%database_name%'
            - { db_username: '%database_user%', db_password: '%database_password%' }
```

Unfortunately when Symfony project is created using Flex there are no such container parameters anymore.

I've tried doing:

```yml
services:
    Symfony\Component\HttpFoundation\Session\Storage\Handler\PdoSessionHandler:
        arguments: ['%env(DATABASE_URL)%']
```

but with `DATABASE_URL` defined in `/.env` as `mysql://user:pass@127.0.0.1:3306/symfony?charset=utf8&serverVersion=5.5` it doesn't work and on attempt to save something to session I'm getting this error:

```
SQLSTATE[3D000]: Invalid catalog name: 1046 No database selected
```

**Comment by sadikoff** [comment]
PdoSessionHandler uses default PDO and that's why default connection string doesn't works. The only way now is to write your own custom session handler to use connection string. You can extend bundled handler and rewrite constructor to parse string and define all vars needed.

**Comment by aik099** [comment]
Either duplicate DB connection data in config OR write only session handler. Both doesn't sound as correct approach.

Hopefully this issue would be addressed with Symfony 4.0 release.

**Comment by javiereguiluz** [comment]
@aik099 thanks for reporting this problem. I agree that it's unfortunate to support incompatible DSN syntaxes. I've opened https://github.com/symfony/symfony/issues/25186 to ask the community to add support to Doctrine DSN in future Symfony versions. So I'm closing this issue until the other one is fixed. Thanks!
